tests/postgres_tests/test_operations.py

```diff
@@ -141,3 +144,42 @@ def test_remove(self):
         self.assertEqual(name, 'RemoveIndexConcurrently')
         self.assertEqual(args, [])
         self.assertEqual(kwargs, {'model_name': 'Pony', 'name': 'pony_pink_idx'})
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', 'PostgreSQL specific tests.')
+@modify_settings(INSTALLED_APPS={'append': 'migrations'})
+class ExtensionMigrationTests(OperationTestBase):
+    app_label = 'test_extention_migration'
+
+    def _database_move(self, operation, direction):
+        project_state = ProjectState()
+        new_state = project_state.clone()
+        with connection.schema_editor() as editor:
+            if direction == 'forwards':
+                operation.database_forwards(self.app_label, editor, project_state, new_state)
+            elif direction == 'backwards':
+                operation.database_backwards(self.app_label, editor, project_state, new_state)
+
+    def test_extention_migration(self):
+        class TestRouter:
+            def allow_migrate(self, db, app_label, **hints):
+                return False
+
+        tests = (
+            ([], 'forwards', 'CREATE EXTENSION', 1),
+            ([TestRouter()], 'forwards', 'CREATE EXTENSION', 0),
+            ([], 'backwards', 'DROP EXTENSION', 1),
```

Comment:
IMO it's enough to test that `CreateExtension` honor `allow_migrate()`, creating extension is already tested in `postgres_tests`.

Reply:
I restored tests when `allow_migrate` is True, it's probably worth to add them.

Reply:
Yes, I think so